### PR TITLE
Refraction

### PR DESCRIPTION
**Pull request by guycalledfrank**
Uses reflection map for refraction. New API:
- material.refraction (float 0-1): defines the amount of visible refraction. Refraction is mixed with diffuse lighting, reflection will be added on top.
- material.refractionIor: defines the amount of distortion based on (outer_IOR / material_IOR). As outer IOR is usually ~1 (air), you can use (1 / materialIor). E.g. for glass it can be (1 / 1.5) which is the default value.

test scene:
http://playcanvas.com/designer/339425/scene/355985

refr

### CODE REVIEW

**Comment by vkalpias** [comment]
We need to add these new material properties to the new editor UI and possibly to the old Designer.

**Comment by willeastcott** [comment]
Is it possible to avoid adding to Designer since it's going away?

File: src/scene/scene_phongmaterial.js [hunk 1]
```diff
@@ -211,6 +211,9 @@ pc.extend(pc, function () {
             this.emissiveMapTint = false;
             this.emissiveIntensity = 1;
 
+            this.refraction = 0;
+            this.refractionIor = 1.0 / 1.5; // approx. (air ior / glass ior)
```
Comment (willeastcott) [inline src/scene/scene_phongmaterial.js]: This property name is completely opaque to me. Suggested alternative: refractionIndex

Comment (guycalledfrank) [inline src/scene/scene_phongmaterial.js]: c'mon, every 3d artist knows what is IOR

Comment (willeastcott) [inline src/scene/scene_phongmaterial.js]: http://en.wikipedia.org/wiki/Refractive_index

File: src/scene/scene_phongmaterial.js [hunk 2]
```diff
@@ -549,7 +556,11 @@ pc.extend(pc, function () {
                 shadingModel:               this.shadingModel,
                 fresnelModel:               this.fresnelModel,
                 packedNormal:               this.normalMap? this.normalMap._compressed : false,
+<<<<<<< HEAD
+                refraction:                 this.refraction,
+=======
```
Comment (willeastcott) [inline src/scene/scene_phongmaterial.js]: Whoops. ;)

Comment (guycalledfrank) [inline src/scene/scene_phongmaterial.js]: oh lol

**Pull request merged**